src/dlgConnectionProfiles.cpp

```diff
@@ -1519,11 +1519,19 @@ void dlgConnectionProfiles::loadProfile(bool alsoConnect)
     if (entries.isEmpty()) {
         firstTimeLoad = true;
     } else {
-        QFile file(qsl("%1%2").arg(folder, profile_history->itemData(profile_history->currentIndex()).toString()));
+        QString fileName{qsl("%1%2").arg(folder, profile_history->itemData(profile_history->currentIndex()).toString())};
+        QFile file(fileName);
         file.open(QFile::ReadOnly | QFile::Text);
         XMLimport importer(pHost);
+
         qDebug() << "[LOADING PROFILE]:" << file.fileName();
-        importer.importPackage(&file, nullptr); // TODO: Missing false return value handler
+        if (auto [success, message] = importer.importPackage(&file, nullptr); !success) {
+            pHost->postMessage(tr("[ ERROR ] - Something went wrong loading your Mudlet profile and it could not be loaded.\n"
```

Comment:
If the call to `XMLimport::importPackage(...)` returns the (translatable), `{ ALERT ] -` message then that is going to be squirted at the `QDebug` output -  you are gonna need a ~~bigger boat~~ *another text* for *that* `return`...!

Reply:
What do you mean?

Reply:
The text assigned to `moanMessage` in the `XMLimport::importPackage` method is in a form appropriate to be posted on the main console - but not to be sent via `qDebug()` to the Linux (at least) OS command line...

Reply:
Ah, this is an intentional design that I came to after a lot of refinement - and one that I think we should adopt elsewhere in the app, at least where relevant. The technical reason why a file save failed will be lost on the vast majority of folks - all they need to know is a) that it didn't work and b) how to fix it, which is what is exactly shown to them. The actual technical details for anyone who'd like to go looking are available too, just in a different spot.